OpenRA.Mods.Common/Traits/Buildings/Refinery.cs

```diff
@@ -84,11 +87,17 @@ public IEnumerable<TraitPair<Harvester>> GetLinkedHarvesters()
 				.Where(a => a.Trait.LinkedProc == self);
 		}
 
-		public bool CanGiveResource(int amount) { return playerResources.CanGiveResources(amount); }
+		public bool CanGiveResource(int amount)
+		{
+			return info.GivesCash ? true : playerResources.CanGiveResources(amount);
```

Comment:
`return info.GivesCash || playerResources.CanGiveResources(amount)` sounds more friendly to me.